### ISSUE TITLE
Plugin chaining not supported with device=MYRIAD?

### ISSUE THREAD

**Issue by fischermario** [openvinotoolkit/dlstreamer_gst#162]
I have tried to run the sample 'face_detection_and_classification' on a RPi and a Jetson Nano (each with a NCSv2 attached). On both machines the startup fails with the following error message:

 ```
ERROR: from element /GstPipeline:pipeline0/GstGvaClassify:gvaclassify0: base_inference plugin intitialization failed
Additional debug info:
/home/ubuntu/gva/gst-video-analytics/gst/inference_elements/base/inference_singleton.cpp(137): acquire_inference_instance (): /GstPipeline:pipeline0/GstGvaClassify:gvaclassify0:
Failed to construct OpenVINOImageInference
	Can not init Myriad device: NC_ERROR

ERROR: pipeline doesn't want to preroll.
Setting pipeline to NULL ...
Freeing pipeline ...
```

If I remove the gvaclassify plugins from the pipeline (leaving only gvadetect in there) it works as expected (btw. the same happens with the Python sample).

Do I need a NCSv2 for every gvaclassify/gvadetect plugin in the pipeline?

Thank you!

**Comment by mliu2020** [comment]
Hi @fischermario,

Can you check #54 for the clue?

There are several issues here:
- The NCS2 will behavior different in host, Docker or Virtual machine, which environment do you on?
- The NCS2 should be installed correctly, you can check it by running OpenVINO only application like I suggested in #161
- The NCS2 seems work well with FP16 format, so please make sure your IR format is aligned.

Mark

**Comment by fischermario** [comment]
Hi @mliu2020,

I have checked #54 for further information. The "solution" of @1000sprites in his last comment was that he removed all gvaclassify elements (which is the same observation that I have made), but that completely defeats the purpose.

I am running the sample directly on the host. There is no Docker or virtualization involved. As suggested I made sure that the sample is using FP16 models instead of FP32. I have verified that the NCSv2 is installed correctly. The error message stays the same.

Are you sure that this sample is supposed to work with only one NCSv2 present?

**Comment by mliu2020** [comment]
Thanks for the quick response,

Could you try to run a native OpenVINO sample and see if it can run normally on NCS2?

Mark

**Comment by fischermario** [comment]
Hi @mliu2020,

the native OpenVINO samples work as expected. I have tested the C++ and Python sample applications.

At first I ran 'hello_query_device'. This is the output:

```
Available devices: 
	Device: MYRIAD
	Metrics: 
		RANGE_FOR_ASYNC_INFER_REQUESTS : { 3, 6, 1 }
		OPTIMIZATION_CAPABILITIES : [ FP16 ]
		DEVICE_THERMAL : UNSUPPORTED TYPE
		SUPPORTED_CONFIG_KEYS : [ DEVICE_ID EXCLUSIVE_ASYNC_REQUESTS LOG_LEVEL VPU_MYRIAD_FORCE_RESET VPU_MYRIAD_PLATFORM VPU_CUSTOM_LAYERS PERF_COUNT VPU_PRINT_RECEIVE_TENSOR_TIME CONFIG_FILE VPU_HW_STAGES_OPTIMIZATION MYRIAD_ENABLE_FORCE_RESET MYRIAD_ENABLE_RECEIVING_TENSOR_TIME MYRIAD_CUSTOM_LAYERS MYRIAD_ENABLE_HW_ACCELERATION ]
		SUPPORTED_METRICS : [ RANGE_FOR_ASYNC_INFER_REQUESTS OPTIMIZATION_CAPABILITIES DEVICE_THERMAL SUPPORTED_CONFIG_KEYS SUPPORTED_METRICS FULL_DEVICE_NAME AVAILABLE_DEVICES ]
		FULL_DEVICE_NAME : Intel Movidius Myriad X VPU
	Default values for device configuration keys: 
		DEVICE_ID : ""
		EXCLUSIVE_ASYNC_REQUESTS : NO
		LOG_LEVEL : LOG_NONE
		VPU_MYRIAD_FORCE_RESET : NO
		VPU_MYRIAD_PLATFORM : ""
		VPU_CUSTOM_LAYERS : ""
		PERF_COUNT : NO
		VPU_PRINT_RECEIVE_TENSOR_TIME : NO
		CONFIG_FILE : ""
		VPU_HW_STAGES_OPTIMIZATION : YES
		MYRIAD_ENABLE_FORCE_RESET : NO
		MYRIAD_ENABLE_RECEIVING_TENSOR_TIME : NO
		MYRIAD_CUSTOM_LAYERS : ""
		MYRIAD_ENABLE_HW_ACCELERATION : YES
```
This is the output of 'object_detection_sample_ssd':
```
[ INFO ] InferenceEngine: 
	API version ............ 2.1
	Build .................. custom_HEAD_f557dca475cb54dcfc9026fbaad0d93ddb85015c
	Description ....... API
Parsing input parameters
[ INFO ] Files were added: 1
[ INFO ]     faceframing.jpg
[ INFO ] Loading Inference Engine
[ INFO ] Device info: 
	MYRIAD
	myriadPlugin version ......... 2.1
	Build ........... custom_HEAD_f557dca475cb54dcfc9026fbaad0d93ddb85015c
[ INFO ] Loading network files:
	intel/face-detection-adas-0001/FP16/face-detection-adas-0001.xml
[ INFO ] Preparing input blobs
[ INFO ] Batch size is 1
[ INFO ] Preparing output blobs
[ INFO ] Loading model to the device
[ INFO ] Create infer request
[ WARNING ] Image is resized from (2400, 1472) to (672, 384)
[ INFO ] Batch size is 1
[ INFO ] Start inference
[ INFO ] Processing output blobs
[0,1] element, prob = 0.999512    (1030,219)-(1495,840) batch id : 0 WILL BE PRINTED!
[1,1] element, prob = 0.0268555    (1647,-45)-(2381,595) batch id : 0
[2,1] element, prob = 0.0239258    (1072,90)-(2078,1180) batch id : 0
[3,1] element, prob = 0.0219727    (28,86)-(1267,1228) batch id : 0
[4,1] element, prob = 0.0200195    (1046,507)-(1386,807) batch id : 0
[5,1] element, prob = 0.0146484    (471,1367)-(709,1474) batch id : 0
[6,1] element, prob = 0.0136719    (1251,-6)-(1422,93) batch id : 0
[7,1] element, prob = 0.0136719    (-143,-117)-(409,558) batch id : 0
[8,1] element, prob = 0.0136719    (2071,945)-(2702,1617) batch id : 0
[9,1] element, prob = 0.0126953    (0,278)-(66,391) batch id : 0
[10,1] element, prob = 0.0126953    (829,819)-(1102,1023) batch id : 0
[11,1] element, prob = 0.0126953    (1517,-140)-(1913,460) batch id : 0
[12,1] element, prob = 0.0126953    (1543,920)-(1885,1704) batch id : 0
[13,1] element, prob = 0.0117188    (0,227)-(65,336) batch id : 0
[14,1] element, prob = 0.0117188    (1298,1012)-(1394,1126) batch id : 0
[15,1] element, prob = 0.0117188    (1191,134)-(1372,288) batch id : 0
[16,1] element, prob = 0.0117188    (1119,198)-(1346,420) batch id : 0
[17,1] element, prob = 0.0117188    (1335,977)-(1476,1134) batch id : 0
[18,1] element, prob = 0.0117188    (475,1248)-(740,1464) batch id : 0
[19,1] element, prob = 0.0117188    (1907,223)-(2083,654) batch id : 0
[20,1] element, prob = 0.0117188    (1338,730)-(1514,1022) batch id : 0
[21,1] element, prob = 0.0117188    (1998,688)-(2400,1414) batch id : 0
[22,1] element, prob = 0.0117188    (-80,-82)-(804,863) batch id : 0
[23,1] element, prob = 0.0107422    (837,891)-(946,1024) batch id : 0
[24,1] element, prob = 0.0107422    (1361,935)-(1457,1080) batch id : 0
[25,1] element, prob = 0.0107422    (1402,995)-(1489,1121) batch id : 0
[26,1] element, prob = 0.0107422    (1475,1033)-(1550,1114) batch id : 0
[27,1] element, prob = 0.0107422    (679,1186)-(754,1304) batch id : 0
[28,1] element, prob = 0.0107422    (1123,-5)-(1318,83) batch id : 0
[29,1] element, prob = 0.0107422    (1144,150)-(1306,306) batch id : 0
[30,1] element, prob = 0.0107422    (1217,193)-(1442,419) batch id : 0
[31,1] element, prob = 0.0107422    (584,1191)-(796,1456) batch id : 0
[32,1] element, prob = 0.0107422    (1014,-9)-(1371,151) batch id : 0
[33,1] element, prob = 0.0107422    (1320,578)-(1515,889) batch id : 0
[34,1] element, prob = 0.0107422    (1395,672)-(1658,909) batch id : 0
[35,1] element, prob = 0.0107422    (628,797)-(868,1032) batch id : 0
[36,1] element, prob = 0.0107422    (604,930)-(895,1139) batch id : 0
[37,1] element, prob = 0.0107422    (1392,-39)-(2085,320) batch id : 0
[38,1] element, prob = 0.0107422    (114,399)-(958,1397) batch id : 0
[ INFO ] Image out_0.bmp created!
[ INFO ] Execution successful

[ INFO ] This sample is an API example, for any performance measurements please use the dedicated benchmark_app tool
```
Should I try something else?

**Comment by fischermario** [comment]
I have found a clue in the Open Model Zoo issue tracker:

https://github.com/openvinotoolkit/open_model_zoo/issues/567#issuecomment-548394044

@mliu2020: Is it possible that the same explanation applies here?

**Comment by mliu2020** [comment]
Hi @fischermario,

Thanks for much for the detailed information and investigation. We have a brief discussion and believe this is a root cause.

We will put more investigation on this and fix it in the upcoming release.

Mark

**Comment by adranit** [comment]
resolved in 2021.4
